commit 4d711942f60198f8ceae164a332a51f18fd67a82
Author: Helio Chissini de Castro <heliocastro@gmail.com>
Date:   Mon Nov 24 15:46:51 2014 -0200

    Fix for bug https://bugs.kde.org/show_bug.cgi?id=329375
    Multihead locking session not properly locks all screens, leaving all
    but first unlocked

diff --git a/ksmserver/screenlocker/greeter/greeterapp.cpp b/ksmserver/screenlocker/greeter/greeterapp.cpp
index ae1c62b..8ac0de4 100644
--- a/ksmserver/screenlocker/greeter/greeterapp.cpp
+++ b/ksmserver/screenlocker/greeter/greeterapp.cpp
@@ -151,8 +151,8 @@ void UnlockApp::desktopResized()
     const bool canLogout = KAuthorized::authorizeKAction("logout") && KAuthorized::authorize("logout");
     const QSet<Solid::PowerManagement::SleepState> spdMethods = Solid::PowerManagement::supportedSleepStates();
     for (int i = m_views.count(); i < nScreens; ++i) {
-        // create the view
-        QDeclarativeView *view = new QDeclarativeView();
+        // create the view - we need create a view per screen in multihead cases
+        QDeclarativeView *view = new QDeclarativeView(desktop()->screen(i));
         connect(view, SIGNAL(statusChanged(QDeclarativeView::Status)),
                 this, SLOT(viewStatusChanged(QDeclarativeView::Status)));
         view->setWindowFlags(Qt::X11BypassWindowManagerHint);
